core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistry.java

```diff
@@ -339,6 +347,31 @@ protected boolean matches(TypeCodec<?> codec, GenericType<?> javaType, boolean i
     }
   }
 
+  protected TypeCodec<TupleValue> createTupleCodec(TupleType tupleType) {
+    // if the attachment point of the input tuple does not match the attachment point
+    // for this registry, adapt it.  This is important to ensure decoding of values uses
+    // the appropriate registry and protocol version.  It has no bearing on serialization,
+    // which will use the tuple type of the input tuple value.
+    if (tupleType.getAttachmentPoint() != attachmentPoint) {
```

Comment:
we want to use the same exact instance, so `!=` is used here, although we could consider using `equals` assuming the associated `AttachmentPoint` implements it.

Reply:
Couldn't you use `tupleType.attach(attachmentPoint)`?

Reply:
Oh, I didn't even realize that was an option at the time.  I just looked at the implementation though and it appears it changes the instance and it's underlying component types, which is problematic, even if we create a new instance as we do here.  So we'd need to find a solution for that.

Reply:
Actually when you initialize `DefaultTupleType` with an `AttachmentPoint` the underlying component type values *are not* updated, which can also be a problem (but it would also be a problem if they were updated in place instead of creating new copies).